app/controllers/application_controller.rb

```diff
@@ -6,8 +6,18 @@ class ApplicationController < ActionController::Base
   def current_user
     @current_user ||= RegisteredUser.find(session[:registered_user_id]) if session[:registered_user_id]
   end
-  
+
   def current_admin
     @current_admin ||= Admin.find(session[:admin_id]) if session[:admin_id]
   end
+
+  def find_user_by_username_in_url(username)
+    pinspiration_credential = PinspirationCredential.find_by(username: username)
+    if pinspiration_credential.present?
+      pinspiration_credential.registered_user
+    else
+      google_credential = GoogleCredential.find_by(username: username)
+      google_credential.registered_user
+    end
+  end
```

Comment:
Nice! Should we make this a helper method? We could certainly use it in the boards controller.